### PR TITLE
fix: properly use dask-awkward optimizations in all scenarios

### PR DESCRIPTION
**Pull request by lgray**
Fixes: https://github.com/dask-contrib/dask-awkward/issues/464

### CODE REVIEW

File: src/dask_histogram/core.py [hunk 1]
```diff
@@ -401,27 +401,28 @@ def optimize(
     keys: Hashable | list[Hashable] | set[Hashable],
     **kwargs: Any,
 ) -> Mapping:
-    if not isinstance(keys, (list, set)):
-        keys = [keys]
```
Comment (martindurant) [inline src/dask_histogram/core.py:405]: The never happened/was uncovered?

Comment (martindurant) [reply src/dask_histogram/core.py:405]: (or maybe `flatten` does the right thing already?)

File: src/dask_histogram/core.py [hunk 2]
```diff
@@ -401,27 +401,28 @@ def optimize(
...
+
+
+def _get_optimization_function():
     # Here we try to run optimizations from dask-awkward (if we detect
     # that dask-awkward has been imported). There is no cost to
     # running this optimization even in cases where it's unncessary
-    # because if no AwkwardInputLayers from daks-awkward are not
+    # because if no AwkwardInputLayers from dask-awkward are
     # detected then the original graph is returned unchanged.
     if dask.config.get("awkward", default=False):
```
Comment (martindurant) [inline src/dask_histogram/core.py:421]: This is populated when dak is imported, correct? This import has to have happened if any dak components are in the graph.

File: tests/test_boost.py [hunk 3]
```diff
@@ -176,6 +176,10 @@ def test_obj_5D_strcat_intcat_rectangular_dak(use_weights):
         dhb.axis.Regular(9, -3.2, 3.2),
         storage=storage,
     )
+
+    # check that we are using the correct optimizer
+    assert h.__dask_optimize__ == dak.lib.optimize.all_optimizations
```
Comment (martindurant) [inline tests/test_boost.py:181]: Because dak is in the CI env; can we mock the case if it missing (config "awkward"=False)

**Review by martindurant — APPROVED** [review]
One minor suggestion

File: src/dask_histogram/core.py (hunk 1 above)
Comment (lgray) [reply src/dask_histogram/core.py:405]: This is copying the function from dask-awkward, which is more thoroughly reviewed and battle hardened.

I know it's a bit cargo-cult, but it is also likely true in this case.

File: src/dask_histogram/core.py (hunk 2 above)
Comment (lgray) [reply src/dask_histogram/core.py:421]: Yes - so the only way this fails is if you're loading a pickled task-graph with dask-awkward stuff in it... But even then I think that requires you've imported the module? So I think this is rather safe.

File: tests/test_boost.py (hunk 3 above)
Comment (lgray) [reply tests/test_boost.py:181]: Conversely, it only matters that it is set correctly when dak is available. No need to test the other case (well, OK, until there are dask-histogram specific optimizations).

**Pull request merged**